src/ar_node_utils.erl

```diff
@@ -226,35 +233,40 @@ start_mining(#{
 								]
 							);
 						false ->
-								ar:report(
-									[
-										could_not_start_mining,
-										{received_invalid_recall_block, FullBlock#block.indep_hash}
-									]
-								)
+							ar:report(
+								[
+									could_not_start_mining,
+									{received_invalid_recall_block, FullBlock#block.indep_hash}
+								]
+							)
 					end
 			end,
 			StateIn;
 		RecallB ->
-			if not is_record(RecallB, block) ->
-				ar:report_console([{erroneous_recall_block, RecallB}]);
-			true ->
-				ar_miner_log:started_hashing(),
-				ar:report([{node_starting_miner, Node}, {recall_block, RecallB#block.height}])
+			case is_record(RecallB, block) of
+				false ->
+					ar:report_console([{erroneous_recall_block, RecallB}]);
+				true ->
+					ar_miner_log:started_hashing(),
+					ar:report([{node_starting_miner, Node}, {recall_block, RecallB#block.height}])
 			end,
-			RecallBFull = make_full_block(
+			case make_full_block(
```

Comment:
This is duplicated in `integrate_new_block`. What about moving this into a function with which takes the recall block shadow? E.g. `store_recall_block_key(RecallBShadow)`

Reply:
✅